packages/node/test/transports/new/https.test.ts

```diff
@@ -0,0 +1,397 @@
+import { createTransport } from '@sentry/core';
+import { EventEnvelope, EventItem } from '@sentry/types';
+import { createEnvelope, serializeEnvelope } from '@sentry/utils';
+import * as http from 'http';
+import * as https from 'https';
+
+import { HTTPModule, HTTPModuleRequestIncomingMessage } from '../../../src/transports/base/http-module';
+// TODO(v7): We're renaming the imported file so this needs to be changed as well
+import { makeNodeTransport } from '../../../src/transports/new';
+import testServerCerts from './test-server-certs';
+
+jest.mock('@sentry/core', () => {
+  const actualCore = jest.requireActual('@sentry/core');
+  return {
+    ...actualCore,
+    createTransport: jest.fn().mockImplementation(actualCore.createTransport),
+  };
+});
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const httpProxyAgent = require('https-proxy-agent');
+jest.mock('https-proxy-agent', () => {
+  return jest.fn().mockImplementation(() => new http.Agent({ keepAlive: false, maxSockets: 30, timeout: 2000 }));
+});
+
+const SUCCESS = 200;
+const RATE_LIMIT = 429;
+const INVALID = 400;
+const FAILED = 500;
+
+interface TestServerOptions {
+  statusCode: number;
+  responseHeaders?: Record<string, string | string[] | undefined>;
+}
+
+let testServer: http.Server | undefined;
+
+function setupTestServer(
+  options: TestServerOptions,
+  requestInspector?: (req: http.IncomingMessage, body: string) => void,
+) {
+  testServer = https.createServer(testServerCerts, (req, res) => {
+    let body = '';
+
+    req.on('data', data => {
+      body += data;
+    });
+
+    req.on('end', () => {
+      requestInspector?.(req, body);
+    });
+
+    res.writeHead(options.statusCode, options.responseHeaders);
+    res.end();
+
+    // also terminate socket because keepalive hangs connection a bit
+    res.connection.end();
+  });
+
+  testServer.listen(8099);
+
+  return new Promise(resolve => {
+    testServer?.on('listening', resolve);
+  });
+}
+
+const TEST_SERVER_URL = 'https://localhost:8099';
+
+const EVENT_ENVELOPE = createEnvelope<EventEnvelope>({ event_id: 'aa3ff046696b4bc6b609ce6d28fde9e2', sent_at: '123' }, [
+  [{ type: 'event' }, { event_id: 'aa3ff046696b4bc6b609ce6d28fde9e2' }] as EventItem,
+]);
+
+const SERIALIZED_EVENT_ENVELOPE = serializeEnvelope(EVENT_ENVELOPE);
+
+const unsafeHttpsModule: HTTPModule = {
```

Comment:
It's unsafe because of the `rejectUnauthorized: false` option. CodeQL and Sonarlint are also complaining but it's fine I think because this is a test.